packages/i18nify-js/src/modules/phoneNumber/utils.ts

```diff
@@ -48,10 +48,17 @@ export const detectCountryAndDialCodeFromPhone = (
 
     // Filter matching countries based on phone number validation regex
     const matchedCountryCode = matchingCountries.find((country) => {
-      const regex = new RegExp(
-        regexMapper[country.countryCode as CountryCodeType],
+      const phoneNumberWithoutDialCode = String(phoneNumber).replace(
```

Comment:
why not using util here `getPhoneNumberWithoutDialCode` ?

Reply:
`getPhoneNumberWithoutDialCode` function uses `getPhoneNumberWithoutDialCode` so if `getPhoneNumberWithoutDialCode` is used inside `getPhoneNumberWithoutDialCode` then it will be an infinite loop.

Reply:
got it